File: Java/Program/Class/Komputer.java
package Class;

public class Komputer extends Barang {
    private Motherboard motherboard;
    private Processor processor;
    private Ram ram;
    private GPU gpu;

    // Default constructor
    public Komputer() {
        super();
        this.motherboard = new Motherboard();
        this.processor = new Processor();
        this.ram = new Ram();
        this.gpu = new GPU();
    }

    // Constructor with raw parameters (composition)
    public Komputer(int id, String nama, int harga, int stok,
                    // Motherboard params
                    int mbId, String mbNama, int mbHarga, int mbStok,
                    String mbNoPart, String mbManufaktur, String mbKondisi,
                    String mbSocket, int mbDDR, String mbChipset,

                    // Processor params
                    int pId, String pNama, int pHarga, int pStok,
                    String pNoPart, String pManufaktur, String pKondisi,
                    String pSocket, int pCore, int pThread, int pTDP,

                    // RAM params
                    int rId, String rNama, int rHarga, int rStok,
                    String rNoPart, String rManufaktur, String rKondisi,
                    int rDDR, int rMemory, int rSpeed,

                    // GPU params
                    int gId, String gNama, int gHarga, int gStok,
                    String gNoPart, String gManufaktur, String gKondisi,
                    int gMemory, int gTDP, int gPcie) {
        super(id, nama, harga, stok);

        this.motherboard = new Motherboard(mbId, mbNama, mbHarga, mbStok,
                                           mbNoPart, mbManufaktur, mbKondisi,
                                           mbSocket, mbDDR, mbChipset);

        this.processor = new Processor(pId, pNama, pHarga, pStok,
                                       pNoPart, pManufaktur, pKondisi,
                                       pSocket, pCore, pThread, pTDP);

        this.ram = new Ram(rId, rNama, rHarga, rStok,
                           rNoPart, rManufaktur, rKondisi,
                           rDDR, rMemory, rSpeed);

        this.gpu = new GPU(gId, gNama, gHarga, gStok,
                           gNoPart, gManufaktur, gKondisi,
                           gMemory, gTDP, gPcie);
    }

    // Getters only (no setters, Komputer fully owns its components)
    public Motherboard getMotherboard() { return motherboard; }
    public Processor getProcessor() { return processor; }
    public Ram getRam() { return ram; }
    public GPU getGPU() { return gpu; }
}

File: Python/Program/Class/Komputer.py
from Class.Barang import Barang
from Class.Motherboard import Motherboard
from Class.Processor import Processor
from Class.Ram import Ram
from Class.GPU import GPU

class Komputer(Barang):
    def __init__(self, id=0, nama="", harga=0, stok=0,
                 # Motherboard params
                 mb_id=0, mb_nama="", mb_harga=0, mb_stok=0,
                 mb_no_part="", mb_manufaktur="", mb_kondisi="",
                 mb_socket="", mb_ddr=0, mb_chipset="",

                 # Processor params
                 p_id=0, p_nama="", p_harga=0, p_stok=0,
                 p_no_part="", p_manufaktur="", p_kondisi="",
                 p_socket="", p_core=0, p_thread=0, p_tdp=0,

                 # RAM params
                 r_id=0, r_nama="", r_harga=0, r_stok=0,
                 r_no_part="", r_manufaktur="", r_kondisi="",
                 r_ddr=0, r_memory=0, r_speed=0,

                 # GPU params
                 g_id=0, g_nama="", g_harga=0, g_stok=0,
                 g_no_part="", g_manufaktur="", g_kondisi="",
                 g_memory=0, g_tdp=0, g_pcie=0):
        super().__init__(id, nama, harga, stok)

        # Composition: Komputer owns these components
        self.motherboard = Motherboard(mb_id, mb_nama, mb_harga, mb_stok,
                                       mb_no_part, mb_manufaktur, mb_kondisi,
                                       mb_socket, mb_ddr, mb_chipset)
        self.processor = Processor(p_id, p_nama, p_harga, p_stok,
                                   p_no_part, p_manufaktur, p_kondisi,
                                   p_socket, p_core, p_thread, p_tdp)
        self.ram = Ram(r_id, r_nama, r_harga, r_stok,
                       r_no_part, r_manufaktur, r_kondisi,
                       r_ddr, r_memory, r_speed)
        self.gpu = GPU(g_id, g_nama, g_harga, g_stok,
                       g_no_part, g_manufaktur, g_kondisi,
                       g_memory, g_tdp, g_pcie)

    # Getters only (no setters → composition = owned by Komputer)
    def get_motherboard(self) -> Motherboard:
        return self.motherboard

    def get_processor(self) -> Processor:
        return self.processor

    def get_ram(self) -> Ram:
        return self.ram

    def get_gpu(self) -> GPU:
        return self.gpu
